### ISSUE TITLE
🐛 [Bug Report] RAF feature not yet fully functional

### ISSUE THREAD

**Issue by korhaldragonir** [cmangos/issues#2546]
## 🐛 Bugreport
I have been searching in the code for quite some time now, and still cannot explain the symptoms I am experiencing.

When I link multiple accounts and then restart the server only the link lowest `referred` account ID works. E.g:

 - referrer: 5, referred: 18 (only this one works)
 - referrer: 5, referred: 81 

Now when I add:

 - referrer: 5, referred: 6 (after a restart only this one works and the referred: 18, stops working)

Was hoping to find the culprit in the code and try to fix it. But unfortunately I am kind of lost.

### Expected behavior
This is the page describing the feature https://wowpedia.fandom.com/wiki/Recruit-A-Friend?oldid=2289753 (It is from Jun 6 - 2010). Since the client 3.3.5a is from Jun 24, 2010 I used this page to try and figure out how it works.

I expected all linked accounts up to, I believe, 10? (needs to be verified, could be lower in 3.3.5a) to gain the the Refer-A-Friend bonus XP so long as all the criteria on the referenced article are met.

### Version & Environment
Client Version: "3.3.5a" (WOTLK)

CMaNGOS Repo & Commit Hash: https://github.com/cmangos/mangos-wotlk/tree/67d01de4c04bc69a81c1a462b2f49a90eb86abae

Database Repo & Commit Hash: https://github.com/cmangos/wotlk-db/tree/b2e13071fa13a6704516c8da79941885338199ab

Operating System: Ubuntu 20.04  (Docker) (Built via https://github.com/korhaldragonir/cmangos-docker/tree/f5a991180a7b23a6c500a3254a4d14eb14279945)

### Steps to reproduce
1. Create 3 accounts and write down the Account ID.
2. Link the first 2, with the first being the referrer.
3. Restart the server.
4. Now when creating a new character for each account and partying them up, both receive the Refer-A-Friend Bonus.
5. Now link the other account to the first account as well.
6. Restart the server.
7. Create a new character on that account as well.
8. Only the first account now benefits from the Refer-A-Friend bonus whilst the 3rd account does not.

I also noticed that when I:
1. Remove the first link.
2. Restart the server.
3. Now the 3rd account and 1st account work as expected. They both get the Refer-A-Friend bonus.

**Comment by korhaldragonir** [comment]
I think it might have something to do with this line: https://github.com/cmangos/mangos-wotlk/commit/627fdc601f8a9140b420b3c3af58c00f3ea58b09#diff-af0402a0555dd224c7d257fe4c346602fd62e42237d11df533ca3fefb7a0f91bR533

In case of the `referred` player it will always only show 1 match. But if you are a recruiter it may have multiple matches, but it will only check the first match, being the one with the lowest Account IDs.

**Comment by killerwife** [comment]
The way I always understood this feature, there can always be only one active recruitee for any recruiter. Thats how it functions in DB right now, in our core and trinitycore. I would need valid proof for this to be in any other way. (and even then its fairly low priority because of how niche this feature is)

**Comment by korhaldragonir** [comment]
I understand. I will check around with some of the people whom I have known to be playing on official back in the days to figure out how it should work. Regarding valid proof, sure I will of course also look into finding the official statements from Blizzard if there are any.

And don't worry I do not expect you to quickly fix it, I am just trying to figure out how it works and how it is implemented. So if it does need to be fixed I can have a whack at it myself.

Just want to find a way to at least contribute a bit to the project since I have been using it since 2007. But since my C skills are limited it takes me quite some time to figure things out.

**Comment by Nytanath** [comment]
Just dropping this here, as I found proof that you could recruit more than one people at a time: https://web.archive.org/web/20100728105829/http://forums.worldofwarcraft.com/thread.html?topicId=8557372932&pageNo=1&sid=1

First post:
> Question: I recently recruited 2 new friends to the game, like 2 weeks ago. When is the starting/cut off for the new benefits? 

I seem to remember from my time on retail WotLK that the limit was either 5 or 10. Not sure about that.

**Comment by Nytanath** [comment]
Yeah I know, just wanted to drop this here as I'd have forgotten otherwise :D
